### ISSUE TITLE
[Bug]Awc client connect websocket ignore the port

### ISSUE THREAD

**Issue by zerodegress** [actix/actix-web#3430]
## Expected Behavior

code like below should perform connection to websocket server correctly

```rust
let (_, conn) = awc::Client::builder().finish().ws("ws://127.0.0.1:9090/").address(addr).connect().await.map_err(|e| {
        println!("{:?}", e);
    }).unwrap()
```

## Current Behavior

![图片](https://github.com/user-attachments/assets/a625e85f-def0-4be8-b423-e8abb970208c)

the server responses 400, here's the wireshark captured
![图片](https://github.com/user-attachments/assets/4f253459-5d81-4e4e-a1eb-53be6652ef9e)

## Possible Solution

noticed that the upgrade request:
![图片](https://github.com/user-attachments/assets/994931c6-c885-4b25-a677-193774c2ed5b)

and the code here in awc ws.rs:
![图片](https://github.com/user-attachments/assets/872421ad-ab73-43f3-a6aa-0f5a9194f4e7)

according to mdn, i thought maybe the port should be appended to the host, like this:
host: 127.0.0.1:9090

## Steps to Reproduce (for bugs)

1.install required dependencies(actix_rt, awc)
2.simply run the code above

## Context

trying to connect the rosbridge_server(ros kinetic) in websocket protocol.

## Your Environment

- Rust Version (I.e, output of `rustc -V`):1.79.0
- Awc Version: 3.5.0

**Comment by zerodegress** [comment]
tempory solution(maybe):
```rust
let (_, conn) = awc::Client::builder().finish().ws("ws://127.0.0.1:9090/").set_header(actix_http::header::HOST, "127.0.0.1:9090").connect().await.map_err(|e| {
        println!("{:?}", e);
    }).unwrap();
```

**Comment by robjtede** [comment]
Seems like `http` uses a non-standard definition of "host" since the `.host()` method actually returns the "hostname".

**Comment by robjtede** [comment]
fixed in 5ad92c0062c3ee972005b9c2b0bbb0c2d5d59fbb

**Comment by robjtede** [comment]
released in https://github.com/actix/actix-web/releases/tag/awc-v3.5.1
